lib/jinterface/java_src/com/ericsson/otp/erlang/AbstractConnection.java

```diff
@@ -1098,12 +1098,12 @@ protected void doConnect() throws IOException, OtpAuthException {
             sendCookie = false;
     }
 
-    protected void doConnect(final String node) throws IOException,
+    protected void doConnect(final OtpPeer peer) throws IOException,
             OtpAuthException {
         try {
-            socket = peer.createTransport(node);
+            socket = peer.createTransport(peer);
             if (traceLevel >= handshakeThreshold) {
-                System.out.println("-> MD5 CONNECT TO " + node);
+                System.out.println("-> MD5 CONNECT TO " + peer.node());
             }
             doConnect();
```

Comment:
The `peer` argument is redundant as the we also have a member variable named `peer` which is the same thing.
I also suggest rename the three doConnect functions, something like this:

doConnect(OtpPeer peer) -> doGenericConnect()
doConnect(int port) -> doPortConnect(int port)
doConnect() -> handshake_we_started()        // this is what the corresponding Erlang function is called in dist_util.erl

Reply:
I have switched the new class to OtpGenericTransportFactory as discussed and I have renamed the three doConnect functions with three different names.

Let me know if you have any other suggestions, then I will squash all the code changes into one commit and the documentation into another one.

I still have to work on adding a test, hopefully in the coming days.